src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java

```diff
@@ -246,6 +246,11 @@ private Target loadBuildSetting(
       if (name.startsWith("no")) {
         potentialStarlarkFlag = potentialStarlarkFlag.substring(2);
       }
+      // Check if the string contains a value, trim off the value if so.
+      int equalsIdx = potentialStarlarkFlag.indexOf('=');
+      if (equalsIdx > 0) {
+        potentialStarlarkFlag = potentialStarlarkFlag.substring(0, equalsIdx);
+      }
```

Comment:
This would break for targets that contain the `=` symbol in their labels. In that case, I'd expect the user to wrap the value around quotes, so that `--//some/flag=packages:my=flag="the flag = value"` is not ambiguous. If it is doable without making the logic too complicated, I'd consider this case as well.

Reply:
It seems like Bazel's option parsers do not expect a label or target that contains `=`:
- https://github.com/bazelbuild/bazel/blob/abddb96338aa2a26e778bfc98a10098885695594/src/main/java/com/google/devtools/common/options/OptionsParserImpl.java#L526-L535
- https://github.com/bazelbuild/bazel/blob/ae07b7ca80048d34ad9558665d51f108d9cb6942/src/main/java/com/google/devtools/build/lib/runtime/StarlarkOptionsParser.java#L148-L153
So, I guess it's probably fine to do the same here?

Reply:
Yes, I guess it should be fine then!